src/main/java/net/anotheria/util/concurrency/SafeIdBasedLockManager.java

```diff
@@ -10,26 +10,32 @@
      * Serialization version unique identifier.
      */
     private static final long serialVersionUID = -7086955847344168761L;
-    private Map<K, IdBasedLock<K>> locks = new HashMap<K, IdBasedLock<K>>();
+    private Map<K, IdBasedLock<K>> locks = new HashMap<>();
 
-    public synchronized IdBasedLock<K> obtainLock(K id) {
-        IdBasedLock<K> lock = locks.get(id);
-        if (lock != null) {
-            lock.increaseRefCount();
+    @Override
+    public IdBasedLock<K> obtainLock(K id) {
+        synchronized (this) {
```

Comment:
Why did you add a synchronized block around this part? This is breaking all the functionality!